### PR TITLE
SPPoint

### PR DESCRIPTION
**Pull request by bar-sc**
Hi,
finished SPPoint section.

### CODE REVIEW

File: SPPoint.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#include <stdio.h>
+
+struct sp_point_t{
+    double *data;
+    int dim;
+    int index;
+};
+
+SPPoint spPointCreate(double *data, int dim, int index) {
+    //Check arguments
```
Comment (RedBird) [inline SPPoint.c]: We should declare the parameters at the head of the method

File: SPPoint.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    //Set point's  parameters
+    point->dim = dim;
+    point->index = index;
+    point->data = newData;
+
+    return point;
+}
+
+SPPoint spPointCopy(SPPoint source) {
+    //Check argument
```
Comment (RedBird) [inline SPPoint.c]: It may be useful to use the "spPointCreate" method with the fields of the "source" argument.

File: SPPoint.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    return (point->data)[axis];
+}
+
+double spPointL2SquaredDistance(SPPoint p, SPPoint q) {
+    //Store the result
+    double result = 0;
+    //Save diff
+    double diff = 0;
+
+    assert (p != NULL || q != NULL || (p->dim) == (q->dim));
```
Comment (RedBird) [inline SPPoint.c]: Is it supposed to be an AND relationship between the conditions? I thinks we should assert them all

**Comment by bar-sc** [comment]
Hi,
update due to your comments.
thanks.

**Pull request merged**